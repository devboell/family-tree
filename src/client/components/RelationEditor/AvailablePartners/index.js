import React from 'react'
import { PropTypes } from 'prop-types'

import Ul from './Ul'
import Li from './Li'

const newPartner = person => ({
  relationId: null,
  partner: {
    id: person.id,
    name: person.name,
  },
  children: [],
})

const AvailablePartners = ({ persons, fields }) =>
  <Ul>
    {persons.map(person => (
      <Li key={`avail-partners-${person.id}`}>
        <button type="button" onClick={() => { fields.push(newPartner(person)) }}>add</button>
        {person.name}
      </Li>
    ))}
  </Ul>

AvailablePartners.propTypes = {
  persons: PropTypes.array.isRequired,
  fields: PropTypes.object.isRequired,
}


export default AvailablePartners
