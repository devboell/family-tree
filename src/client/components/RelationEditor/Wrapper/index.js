import styled from 'styled-components'


const Wrapper = styled.div`
  display: flex;
  flex-basis: 100%;
  flex-flow: row;
  justify-content: space-around;
  border-left: 2px solid gray;
`

export default Wrapper
