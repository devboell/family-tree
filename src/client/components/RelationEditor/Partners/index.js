import React from 'react'
import { PropTypes } from 'prop-types'
import { Field } from 'redux-form'

import ListItem from './ListItem'
import Wrapper from './Wrapper'
import PartnerLabel from './PartnerLabel'
import Ul from './Ul'


const Partners = ({ fields }) =>
  <Wrapper>
    <PartnerLabel>Partners:</PartnerLabel>
    <Ul>
      {fields.map((partner, i) => (
        <Field
          key={`partner-${partner}`}
          name={partner}
          component={ListItem}
          onRemove={() => { fields.remove(i) }}
        />
      ))}
    </Ul>
  </Wrapper>


Partners.propTypes = {
  fields: PropTypes.shape({}).isRequired,
}


export default Partners
