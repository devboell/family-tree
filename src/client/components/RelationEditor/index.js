import React from 'react'
import { PropTypes } from 'prop-types'
import Wrapper from './Wrapper'
import Partners from './Partners'
import AvailablePartners from './AvailablePartners'


const RelationEditor = ({
  fields,
  persons,
}) => (
  <Wrapper>
    <Partners {...{ fields }} />
    <AvailablePartners {...{ fields, persons }} />
  </Wrapper>
)

RelationEditor.propTypes = {
  fields: PropTypes.object.isRequired,
  persons: PropTypes.array.isRequired,
}

export default RelationEditor
