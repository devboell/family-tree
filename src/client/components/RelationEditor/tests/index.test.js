import React from 'react'
import RelationEditor from '../index'

it('Relations, renders without crashing', () => {
  const props = {
    fields: {},
    persons: [],
  }
  expect(shallow(<RelationEditor {...props} />)).toMatchSnapshot()
})
