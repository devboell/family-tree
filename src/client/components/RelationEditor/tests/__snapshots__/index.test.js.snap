// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relations, renders without crashing 1`] = `
<styled.div>
  <Partners
    fields={Object {}}
  />
  <AvailablePartners
    fields={Object {}}
    persons={Array []}
  />
</styled.div>
`;
